hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/ddl/ThriftDDLExecutor.java

```diff
@@ -0,0 +1,254 @@
+        partitionSd.setOutputFormat(sd.getOutputFormat());
+        partitionSd.setSerdeInfo(sd.getSerdeInfo());
+        String fullPartitionPath = FSUtils.getPartitionPath(syncConfig.basePath, partition).toString();
+        List<String> partitionValues = partitionValueExtractor.extractPartitionValuesInPath(partition);
+        partitionSd.setLocation(fullPartitionPath);
+        return new Partition(partitionValues, syncConfig.databaseName, tableName, 0, 0, partitionSd, null);
+      }).collect(Collectors.toList());
+      client.addPartitions(partitionList);
+    } catch (TException e) {
+      LOG.error(syncConfig.databaseName + "." + tableName + " add partition failed", e);
+      throw new HoodieHiveSyncException(syncConfig.databaseName + "." + tableName + " add partition failed", e);
+    }
+  }
+
+  @Override
+  public void updatePartitionsToTable(String tableName, List<String> changedPartitions) {
+    if (changedPartitions.isEmpty()) {
+      LOG.info("No partitions to change for " + tableName);
+      return;
+    }
+    LOG.info("Changing partitions " + changedPartitions.size() + " on " + tableName);
+    try {
+      StorageDescriptor sd = client.getTable(syncConfig.databaseName, tableName).getSd();
+      List<Partition> partitionList = changedPartitions.stream().map(partition -> {
+        Path partitionPath = FSUtils.getPartitionPath(syncConfig.basePath, partition);
+        String partitionScheme = partitionPath.toUri().getScheme();
+        String fullPartitionPath = StorageSchemes.HDFS.getScheme().equals(partitionScheme)
+            ? FSUtils.getDFSFullPartitionPath(fs, partitionPath) : partitionPath.toString();
+        List<String> partitionValues = partitionValueExtractor.extractPartitionValuesInPath(partition);
+        sd.setLocation(fullPartitionPath);
+        return new Partition(partitionValues, syncConfig.databaseName, tableName, 0, 0, sd, null);
+      }).collect(Collectors.toList());
+      client.alterPartitions(syncConfig.databaseName, tableName, partitionList);
+    } catch (TException e) {
+      LOG.error(syncConfig.databaseName + "." + tableName + " update partition failed", e);
+      throw new HoodieHiveSyncException(syncConfig.databaseName + "." + tableName + " update partition failed", e);
+    }
+  }
+
+  @Override
+  public void dropPartitionsToTable(String tableName, List<String> partitionsToDrop) {
+    if (partitionsToDrop.isEmpty()) {
+      LOG.info("No partitions to drop for " + tableName);
+      return;
+    }
+
+    LOG.info("Drop partitions " + partitionsToDrop.size() + " on " + tableName);
+    try {
+      client.dropPartitions(syncConfig.databaseName, tableName, partitionsToDrop);
+    } catch (TException e) {
+      LOG.error(syncConfig.databaseName + "." + tableName + " drop partition failed", e);
+      throw new HoodieHiveSyncException(syncConfig.databaseName + "." + tableName + " drop partition failed", e);
+    }
+  }
+
+  @Override
+  public void close() {
```

Comment:
this class for the most part has overlap with HMSDDLExecutor. Can we create a base class and inherit in both executors. you can introduce abstract methods just for the client specific calls. lets try to reuse code as much as possible.

Reply:
hey @nsivabalan can you suggest how I should be restructuring the classes? Thanks.